homeassistant/components/light/device_automation.py

```diff
@@ -0,0 +1,42 @@
+"""Provides device automations for lights."""
+import voluptuous as vol
+
+import homeassistant.components.automation.state as state
+from homeassistant.const import CONF_ENTITY_ID, CONF_PLATFORM, CONF_TYPE
+import homeassistant.helpers.config_validation as cv
+from . import DOMAIN
+
+ENTITY_TRIGGERS = [
+    {
+        # Trigger when light is turned on
+        CONF_PLATFORM: '.{}.device_automation'.format(DOMAIN),
+        CONF_TYPE: 'turn_on',
+    },
+    {
+        # Trigger when light is turned off
+        CONF_PLATFORM: '.{}.device_automation'.format(DOMAIN),
+        CONF_TYPE: 'turn_off',
+    },
+]
+
+TRIGGER_SCHEMA = vol.All(vol.Schema({
+    vol.Required(CONF_PLATFORM): '.{}.device_automation'.format(DOMAIN),
+    vol.Required(CONF_ENTITY_ID): cv.entity_ids,
+    vol.Required(CONF_TYPE): str,
+}))
+
+
+async def async_trigger(hass, config, action, automation_info):
```

Comment:
Name it `async_attach_trigger`. What is config vs automation_info ? Shouldn't we just have `hass, trigger_config, action` ?

Reply:
Ok, but I leave it for now so the new trigger can be tested without any changes to the existing automation framework.